code/game/objects/items/implants/implant_mindshield.dm

```diff
@@ -20,20 +20,22 @@
 	. = ..()
 	if(!.)
 		return FALSE
-	ADD_TRAIT(target, TRAIT_MINDSHIELD, IMPLANT_TRAIT)
 	if(target.mind)
-		if(target.mind.unconvertable)
+		var/flags = SEND_SIGNAL(target.mind, COMSIG_MINDSHIELD_IMPLANTED, user)
+		if(flags & COMPONENT_MINDSHIELD_RESIST || target.mind.unconvertable)
```

Comment:
The check for failure to implant should be separate from the main signal. So like, make a new comsig, call it COMSIG_PRE_MINDSHIELD_IMPLANT or something, and have the headrev register to/respond to that

Reply:
Why should it check beforehand? Wouldn't it just reveal headrevs the moment you start mindshielding, rather than once they are mindshielded?

Reply:
If there's a signal that can prevent mindshielding, it should be done before we send the signal that other stuff uses to tell the user things like "You've been converted!"